src/main/kotlin/test/ktortemplate/conf/database/DatabaseConnection.kt

```diff
@@ -9,7 +12,14 @@ class DatabaseConnection(private val dataSource: DataSource) {
         Database.connect(dataSource)
     }
 
-    fun <T> query(block: () -> T): T = transaction(database) {
+    fun <T> query(block: Transaction.() -> T): T = transaction(database) {
+        block()
+    }
+
+    suspend fun <T> suspendedQuery(block: suspend Transaction.() -> T): T = newSuspendedTransaction(Dispatchers.Default, database) {
```

Comment:
Added the new co-routine friendly wrapper. In this case we're using the Dispatchers.Default (i.e., it is backed by a shared pool of threads on JVM. By default, the maximal level of parallelism used by this dispatcher is equal to the number of CPU cores, but is at least two).

I'm a bit on the fence for this one, because a good default might be `Dispatchers.IO` instead. What do you think @monxalo ?

Reply:
In practical terms, `Dispatchers.IO` has a larger thread pool and elastic which scales based on usage, and is best suited for IO operations which is basically want we are doing here, so i agree the best default should be `Dispatchers.IO`.